api/datastore/services.go

```diff
@@ -0,0 +1,581 @@
+	return store.TeamMembershipService
+}
+
+// Team gives access to the Team data management layer
+func (store *Store) Team() dataservices.TeamService {
+	return store.TeamService
+}
+
+// TunnelServer gives access to the TunnelServer data management layer
+func (store *Store) TunnelServer() dataservices.TunnelServerService {
+	return store.TunnelServerService
+}
+
+// User gives access to the User data management layer
+func (store *Store) User() dataservices.UserService {
+	return store.UserService
+}
+
+// Version gives access to the Version data management layer
+func (store *Store) Version() dataservices.VersionService {
+	return store.VersionService
+}
+
+// Webhook gives access to the Webhook data management layer
+func (store *Store) Webhook() dataservices.WebhookService {
+	return store.WebhookService
+}
+
+type storeExport struct {
+	CustomTemplate     []portainer.CustomTemplate     `json:"customtemplates,omitempty"`
+	EdgeGroup          []portainer.EdgeGroup          `json:"edgegroups,omitempty"`
+	EdgeJob            []portainer.EdgeJob            `json:"edgejobs,omitempty"`
+	EdgeStack          []portainer.EdgeStack          `json:"edge_stack,omitempty"`
+	Endpoint           []portainer.Endpoint           `json:"endpoints,omitempty"`
+	EndpointGroup      []portainer.EndpointGroup      `json:"endpoint_groups,omitempty"`
+	EndpointRelation   []portainer.EndpointRelation   `json:"endpoint_relations,omitempty"`
+	Extensions         []portainer.Extension          `json:"extension,omitempty"`
+	HelmUserRepository []portainer.HelmUserRepository `json:"helm_user_repository,omitempty"`
+	Registry           []portainer.Registry           `json:"registries,omitempty"`
+	ResourceControl    []portainer.ResourceControl    `json:"resource_control,omitempty"`
+	Role               []portainer.Role               `json:"roles,omitempty"`
+	Schedules          []portainer.Schedule           `json:"schedules,omitempty"`
+	Settings           portainer.Settings             `json:"settings,omitempty"`
+	SSLSettings        portainer.SSLSettings          `json:"ssl,omitempty"`
+	Stack              []portainer.Stack              `json:"stacks,omitempty"`
+	Tag                []portainer.Tag                `json:"tags,omitempty"`
+	TeamMembership     []portainer.TeamMembership     `json:"team_membership,omitempty"`
+	Team               []portainer.Team               `json:"teams,omitempty"`
+	TunnelServer       portainer.TunnelServerInfo     `json:"tunnel_server,omitempty"`
+	User               []portainer.User               `json:"users,omitempty"`
+	Version            map[string]string              `json:"version,omitempty"`
+	Webhook            []portainer.Webhook            `json:"webhooks,omitempty"`
+}
+
+func (store *Store) Export(filename string) (err error) {
+
+	backup := storeExport{}
+
+	if c, err := store.CustomTemplate().CustomTemplates(); err != nil {
+		logrus.WithError(err).Debugf("Export boom")
```

Comment:
maybe a more descriptive message?
also, why don't return the error?

Reply:
need to check

Reply:
was this checked?